cmk/base/plugins/agent_based/cisco_entity_sensors.py

```diff
@@ -0,0 +1,134 @@
+# .1.3.6.1.4.1.9.9.91.1.1.1.1.5.47 = INTEGER: 1                                  # entSensorStatus (ok)
+# .1.3.6.1.4.1.9.9.91.1.1.1.1.6.47 = Timeticks: (397951199) 46 days, 1:25:11.99  # entSensorValueTimeStamp
+# .1.3.6.1.4.1.9.9.91.1.1.1.1.7.47 = INTEGER: 60                                 # entSensorValueUpdateRate (seconds)
+# .1.3.6.1.4.1.9.9.91.1.1.1.1.8.47 = INTEGER: 10            # entSensorMeasuredEntity -> entPhysicalIndex -> no match
+#
+# CISCO-ENTITY-SENSOR-MIB::entSensorType.47 = INTEGER: rpm(10)
+# CISCO-ENTITY-SENSOR-MIB::entSensorScale.47 = INTEGER: units(9)
+# CISCO-ENTITY-SENSOR-MIB::entSensorPrecision.47 = INTEGER: 0
+# CISCO-ENTITY-SENSOR-MIB::entSensorValue.47 = INTEGER: 2820
+# CISCO-ENTITY-SENSOR-MIB::entSensorStatus.47 = INTEGER: ok(1)
+# CISCO-ENTITY-SENSOR-MIB::entSensorValueTimeStamp.47 = Timeticks: (397951199) 46 days, 1:25:11.99
+# CISCO-ENTITY-SENSOR-MIB::entSensorValueUpdateRate.47 = INTEGER: 60 seconds
+# CISCO-ENTITY-SENSOR-MIB::entSensorMeasuredEntity.47 = INTEGER: 10
+#
+# entPhysicalIndex
+# .1.3.6.1.2.1.47.1.1.1.1.7.9 = STRING: 'CHASSIS-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.10 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.11 = STRING: 'FAN-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.12 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.13 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.14 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.15 = STRING: 'PSU-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.16 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.17 = STRING: 'MEMORY-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.18 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.19 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.20 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.21 = STRING: 'SSD-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.22 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.23 = STRING: 'CPU-1'
+# .1.3.6.1.2.1.47.1.1.1.1.7.24 = ''
+# .1.3.6.1.2.1.47.1.1.1.1.7.25 = ''
+
+from typing import List
+
+from cmk.base.plugins.agent_based.agent_based_api.v1 import (
+    all_of,
+    contains,
+    register,
+    OIDEnd,
+    startswith,
+    SNMPTree,
+)
+from cmk.base.plugins.agent_based.agent_based_api.v1.type_defs import (
+    StringTable,
+)
+
+from cmk.base.plugins.agent_based.utils.entity_sensors import (
+    OIDSysDescr,
+    ENTITY_SENSOR_TYPES,
+    ENTITY_SENSOR_SCALING,
+    EntitySensor,
+    EntitySensorSection,
+    sensor_status_descr,
+    sensor_state,
+    unit_from_device_unit,
+    reformat_sensor_name,
+)
+
+
+def parse_cisco_entity_sensors(string_table: List[StringTable]) -> EntitySensorSection:
+    section: EntitySensorSection = {}
+    sensor_names = {i[0]: i[1] for i in string_table[0]}
+    for oid_end, sensor_type_nr, scaling_nr, reading, status_nr, device_unit in string_table[1]:
+        # Some devices support the ENTITY-MIB including sensor/entity names.
+        # Others (e.g. Cisco Firepower FPR-1140 Security Appliance) do not support
+        # this MIB, thus we use OID as item instead
+
+        # do not add undefined and temp sensors (duplicate with cisco_temperature)
+        if sensor_type_nr not in ['0', '8']:
```

Comment:
We have quite some duplicate code here. It would be better to define a generic parse function in utils/entity_sensors.py which takes an additional input argument 
`sensor_types_ignore: Container[str]`
You can then use this function for both snmp sections, for example
```
def parse_entity_sensors(string_table: List[StringTable]):
    return utils.parse_entity_sensors(string_table, set())
```